baseapp/baseapp.go

```diff
@@ -740,15 +716,15 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte) (gInfo sdk.GasInfo, re
 		if mode == runTxModeDeliver {
 			// When block gas exceeds, it'll panic and won't commit the cached store.
 			consumeBlockGas()
-
-			msCache.Write()
```

Comment:
When is this write called now?

Reply:
Part of `RunAtomic`, `RunAtomic` will commit when success, discard when fail.

Reply:
RunAtomic itself doesn't call Write, as far as I can see?

Reply:
> RunAtomic itself doesn't call Write, as far as I can see?

The semantic of commit is implemented in RunAtomic directly, there's no write method anymore.

Reply:
I don't understand what you mean. runTx must call msCache.Write, at some point, if mode == runTxModeDeliver. Where does that happen now?

Reply:
> I don't understand what you mean. runTx must call msCache.Write, at some point, if mode == runTxModeDeliver. I see that the Write call here is deleted, but I don't see where another Write call is made. Where does that Write happen, now?

`msCache` is gone as well :)

Reply:
I understand, that's fine, but it doesn't address my point. My point is that runTx operates on a fork of app state, and (assuming mode == runTxModeDeliver) that state must be committed if the runTx operation succeeds, and must be discarded if the runTx operation fails. Where is that logic now?